packages/workbox-window/src/utils/WorkboxEvent.ts

```diff
@@ -9,36 +9,34 @@
 import {WorkboxEventTarget} from './WorkboxEventTarget.js';
 import '../_version.js';
 
+type PropExtractor<WorkboxEventInterface> = Omit<WorkboxEventInterface, "type">;
 
 /**
  * A minimal `Event` subclass shim.
  * This doesn't *actually* subclass `Event` because not all browsers support
  * constructable `EventTarget`, and using a real `Event` will error.
  * @private
  */
-export class WorkboxEvent {
+export class WorkboxEvent<K extends keyof WorkboxEventMap> {
   target?: WorkboxEventTarget;
   sw: ServiceWorker;
 
-  constructor(public type: keyof WorkboxEventMap, props: Object) {
+  constructor(public type: K, props: PropExtractor<WorkboxEventMap[K]>) {
     Object.assign(this, props);
   }
 }
 
-export interface WorkboxMessageEvent extends WorkboxEvent {
-  type: 'message';
+export interface WorkboxMessageEvent extends WorkboxEvent<"message"> {
   originalEvent: Event;
   data: any;
 }
 
-export interface WorkboxLifecycleEvent extends WorkboxEvent {
-  type: keyof WorkboxLifecycleEventMap;
-  originalEvent: Event;
+export interface WorkboxLifecycleEvent extends WorkboxEvent<keyof WorkboxLifecycleEventMap> {
+  originalEvent?: Event;
```

Comment:
I had to add ? because we don't pass originalEvent here:
https://github.com/GoogleChrome/workbox/blob/fec46efbc3ed02eaf0e10d3f65ef11662e9e0a4e/packages/workbox-window/src/Workbox.ts#L142-L145